File: packages/frontend/src/api/notification.api.ts
import api from './client';
import type { ApiResponse, Notification } from '@/types';

export const notificationApi = {
  list: (unread = false) =>
    api.get<ApiResponse<Notification[]>>('/notifications', { params: { unread } }).then((r) => r.data.data),

  unreadCount: () =>
    api.get<ApiResponse<{ count: number }>>('/notifications/unread-count').then((r) => r.data.data),

  markRead: (id: string) =>
    api.put<ApiResponse<unknown>>(`/notifications/${id}/read`).then((r) => r.data),

  markAllRead: () =>
    api.put<ApiResponse<unknown>>('/notifications/read-all').then((r) => r.data),
};

File: packages/mobile/lib/core/api/notification_api.dart
import 'package:dio/dio.dart';

import '../network/api_response.dart';

class NotificationApi {
  NotificationApi(this._dio);
  final Dio _dio;

  Future<List<dynamic>> list({bool unread = false}) async {
    final res = await _dio.get<Map<String, dynamic>>(
      '/notifications',
      queryParameters: {'unread': unread},
    );
    return ApiResponse.unwrapData(res.data!, (d) => d as List<dynamic>);
  }

  Future<Map<String, dynamic>> unreadCount() async {
    final res = await _dio.get<Map<String, dynamic>>('/notifications/unread-count');
    return ApiResponse.unwrapData(res.data!, (d) => d as Map<String, dynamic>);
  }

  Future<void> markRead(String id) async {
    await _dio.put<Map<String, dynamic>>('/notifications/$id/read');
  }

  Future<void> markAllRead() async {
    await _dio.put<Map<String, dynamic>>('/notifications/read-all');
  }
}
